check-submodule.py

```diff
@@ -66,23 +68,76 @@
     ],
     "tree-sitter-python": ["*.py"],
 }
+
+
+class JsonDiff:
+    def __init__(
+        self,
+        old_metrics: T.List[pathlib.Path],
+        new_metrics: T.List[pathlib.Path],
+        compare_dir: pathlib.Path,
+        max_workers: int,
+    ):
+        self.compare_dir = compare_dir
+        self.max_workers = max_workers
+
+        # Max number of filepaths in a sublist
+        n = math.ceil(len(old_metrics) / max_workers)
+
+        # Assign a certain number of filepaths to each worker
+        self.workers_filepaths = [
+            zip(old_metrics[i * n : (i + 1) * n], new_metrics[i * n : (i + 1) * n])
+            for i in range((len(old_metrics) + n - 1) // n)
+        ]
+
+    # Run asynchronous comparisons between json files.
+    async def diff(self):
+        # Define the max number of coroutines used to compare json files
+        all_coroutines = asyncio.gather(
+            *[
+                self._worker(worker_filepaths)
+                for worker_filepaths in self.workers_filepaths
+            ]
+        )
+
+        await all_coroutines
+
+    # Save json files of differences and minimal tests in the chosen directory.
+    async def _worker(self, worker_list: T.List[pathlib.Path]):
+        for old_filename, new_filename in worker_list:
+
+            # Run json-diff asynchronously
+            proc = await asyncio.create_subprocess_shell(
+                f"json-diff -j {old_filename} {new_filename}",
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+
+            # Wait json-diff to terminate
+            await proc.wait()
+
+            # If two json files are identical, skip to the next pair
+            if proc.returncode == 0:
+                continue
+
+            # Get json-diff output
+            stdout, _ = await proc.communicate()
+
+            # Convert bytes in an unicode string
+            json_output = stdout.decode("utf-8")
+
+            # Dump json file of differences
+            dump_json_file(json_output, new_filename, self.compare_dir)
+
+            # Compute minimal tests
+            compute_minimal_tests(old_filename, new_filename, self.compare_dir)
+
+
 # Run a subprocess.
 def run_subprocess(cmd: str, *args: T.Union[str, pathlib.Path]) -> None:
     subprocess.run([cmd, *args])
```

Comment:
I wonder why git shows this as new. Could you move it to the top, so maybe it'll be easier for git to figure out it's still the same?